app/src/androidTest/java/org/kiwix/kiwixmobile/search/SearchRobot.kt

```diff
@@ -32,7 +37,12 @@ class SearchRobot : BaseRobot() {
   fun clickOnSearchItemInSearchList() {
     BaristaSleepInteractions.sleep(TestUtils.TEST_PAUSE_MS_FOR_SEARCH_TEST.toLong())
     isVisible(ViewId(R.id.search_list))
-    clickOn(ViewId(R.id.list_item_search_text))
+    onView(withId(R.id.search_list)).perform(
+      actionOnItemAtPosition<RecyclerView.ViewHolder>(
+        0,
+        click()
```

Comment:
@gouri-panda, Yes there is a difference, we are clicking on the first item of the `recycler view` since we are using the `SuggestionSearch` so it gives us more results, and here we want to click on the first item to test `search` functionality. This approach is good for future test cases if we change the zim file and if that gives results more than one then it will only click on the first item of that search list to properly test the functionality. In the previous test case, we assumed there was only one result coming from the search and we were directly clicking on the `recycler view item id`.